### PR TITLE
feat(ses): DedicatedIpPool, ConfigurationSet and EmailIdentity

### PR DESCRIPTION
**Pull request by jogold**
Add L2 constructs for `DedidactedIpPool`, `ConfigurationSet` and `EmailIdentity`.

`EmailIdentity` can be created for a hosted zone. In this case, all the necessary
DKIM and MAIL FROM records are created automatically.

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by corymhall — CHANGES_REQUESTED** [review]
@jogold I took a first pass and only have some minor things. I'll be able to do a closer review next week.

File: packages/@aws-cdk/aws-ses/lib/email-identity.ts [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+  RSA_1024_BIT = 'RSA_1024_BIT',
+
+  /**
+   * RSA 2048-bit
+   */
+  RSA_2048_BIT = 'RSA_2048_BIT'
+}
+
+/**
+ * An email identity
```
[the remark this answers is not in the document]
Comment (jogold) [reply packages/@aws-cdk/aws-ses/lib/email-identity.ts:299]: Isn't this required only when the name doesn't match with the real CF resource? (I think there's a aws lint rule for this). Here we have `EmailIdentity` matching `AWS::SES::EmailIdentity`.

File: packages/@aws-cdk/aws-ses/lib/email-identity.ts [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+  }
+
+  /**
+   * Bring Your Own DKIM
+   *
+   * @param options Options for BYOD DKIM
+   *
+   * @see https://docs.aws.amazon.com/ses/latest/dg/send-email-authentication-dkim-bring-your-own.html
+   */
+  public static byodDkim(options: ByodDkimOptions): DkimIdentity {
```
[the remark this answers is not in the document]
Comment (jogold) [reply packages/@aws-cdk/aws-ses/lib/email-identity.ts:195]: Yes, good catch.

File: packages/@aws-cdk/aws-ses/lib/email-identity.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jogold) [reply packages/@aws-cdk/aws-ses/lib/email-identity.ts:308]: https://github.com/aws/aws-cdk/blob/5ef106b9fbfcaccb0d22b84feebc79b59ff7eea0/packages/awslint/lib/rules/resource.ts#L58

File: packages/@aws-cdk/aws-ses/README.md [hunk 3]
```diff
@@ -112,3 +112,91 @@ new ses.AllowListReceiptFilter(this, 'AllowList', {
...
+
+new ses.ConfigurationSet(this, 'ConfigurationSet', {
+  customTrackingRedirectDomain: 'track.cdk.dev',
+  suppressionReasons: ses.SuppressionReasons.COMPLAINTS_ONLY,
+  tlsPolicy: ses.ConfigurationSetTlsPolicy.REQUIRE,
+  dedicatedIpPool: myPool,
+});
+```
+
+### Email identity
```
Comment (corymhall) [inline packages/@aws-cdk/aws-ses/README.md:149]: Just a question, do you know if CloudFormation waits for the verification or does it just submit
the request and return without waiting? I guess the question is can you create an `EmailIdentity`
without the identity ever getting verified?

**Review by corymhall — COMMENTED** [review]
@jogold everything looks great, thanks for the super high quality PR! I just have 2 minor
questions/comments on the readme and then we are good to go.

File: packages/@aws-cdk/aws-ses/README.md (hunk 3 above)
Comment (jogold) [reply packages/@aws-cdk/aws-ses/README.md:149]: CF doesn't wait, so yes you can create an identity that remains unverified. This allows to run integ tests 🙂

**Pull request merged**